02_vc/assignment/henke/MatchingCardGame/MatchingCardGame/MatchingCardViewController.swift

```diff
@@ -0,0 +1,130 @@
+//
+//  MatchingCardViewController.swift
+//  MatchingCardGame
+//
+//  Created by Student on 16.10.18.
+//  Copyright © 2018 Alexander Dobrynin. All rights reserved.
+//
+
+import UIKit
+
+class MatchingCardViewController: UIViewController {
+
+    @IBOutlet var cardButtons: [UIButton]!
+    @IBOutlet weak var scoreLabel: UILabel!
+    
+    var gameCards = [Card]()
+    
+    var pendingCard: Card?
+    
+    var globalScore = 0 {
+        didSet {
+            scoreLabel.text = "Score: \(globalScore)"
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        globalScore = 0
+        
+        var deck = Deck()
+        let numberOfCards = cardButtons.indices
+        
+        for _ in numberOfCards { // matching cardButtons with gameCards (1 to 1)
+            if let card = deck.drawRandomCard() {
+                gameCards.append(card)
+            }
+        }
+        
+        assert(gameCards.count == cardButtons.count)
+    }
+
+    // first turn: pending
+    // if pending: match or no match
+    // repeat
+    
+    @IBAction func flipCard(_ sender: UIButton) {
+        let index = cardButtons.firstIndex(of: sender)!
+        let card = gameCards[index]
+        
+        if let pending = pendingCard {
+            if pendingCard(pending, isMatchingWith: card) {
+                flip(sender, to: card)
+                Timer.scheduledTimer(withTimeInterval: 1, repeats: false) { _ in
+               // self.hideButton(sender, to: card)
+                    sender.isHidden = true
+                    let pendingButton = self.cardButtons.first(where: {(button: UIButton) -> Bool in
+                        return button.currentTitle == pending.description
+                    })
+                    pendingButton?.isEnabled = false
```

Comment:
Es müssten beide Buttons "disabled" werden, nicht nur der PendingButton. Aber hier lieber `isUserInteractionEnabled` verwenden! Das andere funktionier in diesem Anwendungsfall nicht